doc/conf.py

```diff
@@ -454,13 +454,41 @@ def generate_min_dependency_substitutions(app):
         f.write(output)
 
 
+def check_min_dependencies_readme(app):
+    """Check min dependencies for README.rst."""
+    from sklearn._build_utils.min_dependencies import dependent_packages
+
+    pattern = re.compile(r'(\.\. \|)' +
+                         r'(([A-Za-z]+\-?)+)' +
+                         r'(MinVersion\| replace::)' +
+                         r'( [0-9]+\.[0-9]+(\.[0-9]+)?)')
+
+    with (Path('..') / 'README.rst').open('r') as f:
+        for line in f:
+            if pattern.match(line):
+                dependency = pattern.sub(r"\2\5", line)
+                (package, version) = dependency.lower().split(" ")
+
+                # Avoid non-defined packages
+                if package in dependent_packages:
+                    version = parse(version)
+                    min_version = parse(dependent_packages[package][0])
+
+                    if version != min_version:
+                        raise RuntimeError(f"{package} {min_version} was "
+                                           f"expected in README.rst file "
+                                           f"but {package} {version} was "
+                                           "found.")
+
+
 # Config for sphinx_issues
 
 # we use the issues path for PRs since the issues URL will forward
 issues_github_path = 'scikit-learn/scikit-learn'
 
 
 def setup(app):
+    app.connect('builder-inited', check_min_dependencies_readme)
```

Comment:
Having sphinx check this is a little strange because I do not think sphinx needs `README.rst` to generate the docs.

This feels like we are linting `README.rst`, so maybe we can have a `build_tools/check_readme.py` and call it in `build_tools/linting.sh`?

Reply:
Why not just have this as `sklearn/tests/test_min_dependencies_readme.py`?

We already have `sklearn/tests/test_docstring_parameters.py` which is a related kind of check

Reply:
> Why not just have this as sklearn/tests/test_min_dependencies_readme.py?

I am -0.5 for this approach. The test would be required to go up two directories to reach the `README.rst` and it would be testing something "outside the sklearn module".

Reply:
fair, but FWIW we already do that in `sklearn/tests/test_common.py/test_configure()` which accesses the root `setup.py`

Reply:
Oh wow, I didn't know we had an `exec` in our code base.

Since we already have something like this, I am okay with having a `sklearn/tests/test_min_dependencies_readme.py`.

Reply:
I am also okay with having `sklearn/tests/test_min_dependencies_readme.py`.

Thank you @NicolasHug and @thomasjpfan for your suggestions!